File: Haskell/Primes.hs
module Primes where

primes = 2 : primes'
  where
    primes' = sieve [3,5..] 9 primes'
    sieve (x:xs) q ps@ ~(p:t)
     | x < q = x : sieve xs q ps
     | True  =     sieve [x | x <- xs, rem x p /= 0] (head t^2) t

largestPrimeFactor n =
   last [ x | x <- takeWhile (< (ceiling . sqrt . fromIntegral) n) primes, n `rem` x == 0 ]

isPrime x = elem x [ y | y <- takeWhile (<=x) primes ]

File: Rust/src/primes.rs
pub struct Primes {}

fn is_prime(n: u64) -> bool {
    if n == 2 {
        return true;
    }

    if n % 2 == 0 {
        return false;
    }

    let limit = f64::sqrt(n as f64) as u64 + 1;
    for x in 3..limit {
        if n % x == 0 {
            return false;
        }
    }

    true
}

pub fn nth_prime(n: u64) -> u64 {
    if n == 1 {
        return 2;
    }

    let mut count = 1;
    let mut candidate = 1;

    while count < n {
        candidate += 2;
        if is_prime(candidate) {
            count += 1;
        }
    }

    candidate
}
